src/app/main/ErrorBoundry.tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react"
+
+export class ErrorBoundary extends React.Component<{onError: (err : Error) => void}, {hasError: boolean}> {
+  constructor(props : any) {
+    super(props)
+    this.state = { hasError: false }
+  }
+
+  public componentDidCatch(error : Error) {
+    this.setState({ hasError: true })
+    this.props.onError(error)
```

Comment:
Is this onError function dependent on state in anyway? If so you should consider moving line 11 into a callback function of the `setState` method on line 10.

Reply:
not dependent, as such keeping as is for now